spec/lib/tasks/import_product_images_rake_spec.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+require 'rake'
+
+RSpec.describe 'ofn:import:product_images' do
+  before do
+    Rake.application.load_rakefile
+    Rake::Task.define_task(:environment)
+    Rake::Task['ofn:import:product_images'].reenable
+  end
+
+  after do
+    Rake::Task['ofn:import:product_images'].clear
+  end
+
+  describe 'task' do
+    context "filename is blank" do
+      it 'raises an error' do
+        expect {
+          Rake::Task['ofn:import:product_images'].invoke('')
+        }.to raise_error(RuntimeError,
+                         'Filename required')
+      end
+    end
+
+    context "invalid CSV format" do
+      it 'raises an error if CSV columns are missing' do
+        allow(CSV).to receive(:read).and_return(CSV::Table.new([]))
+        Rake::Task['ofn:import:product_images'].reenable
+
+        expect {
+          Rake::Task['ofn:import:product_images'].invoke('path/to/csv/file.csv')
+        }.to raise_error(RuntimeError, 'CSV columns reqired: ["producer", "name", "image_url"]')
+      end
+    end
+
+    context "valid CSV" do
+      it 'imports images for each product in the CSV that exists and does not have images' do
+        filename = 'path/to/csv/file.csv'
+
+        csv_data = [
+          { producer: 'Producer 1', name: 'Product 1', image_url: 'http://example.com/image1.jpg' },
+          { producer: 'Producer 2', name: 'Product 2', image_url: 'http://example.com/image2.jpg' },
+          { producer: 'Producer 3', name: 'Product 3', image_url: 'http://example.com/image3.jpg' }
+        ]
+
+        csv_rows = csv_data.map do |hash|
+          CSV::Row.new(hash.keys, hash.values)
+        end
+
+        csv_table = CSV::Table.new(csv_rows)
+
+        allow(CSV).to receive(:read).and_return(csv_table)
+
+        allow(Enterprise).to receive(:find_by!).with(name: 'Producer 1').and_return(double)
+        allow(Enterprise).to receive(:find_by!).with(name: 'Producer 2').and_return(double)
+        allow(Enterprise).to receive(:find_by!).with(name: 'Producer 3').and_return(double)
+
+        allow(Spree::Product).to receive(:where).and_return(
+          double(first: nil), double(first: double(image: nil)), double(first: double(image: true))
+        )
```

Comment:
It's better to use `instance_double` , `instance_double` will check if method actually exist on the class.

Reply:
Good point @rioug, updated. Thanks!